arch/arm/src/stm32h7/Kconfig

```diff
@@ -284,11 +284,24 @@ config STM32H7_HAVE_SMPS
 	bool
 	default n
 
+choice
+	prompt "STM32 H7 Power Supply Selection"
+	default STM32H7_PWR_DIRECT_SMPS_SUPPLY
```

Comment:
I do not think the default is correct.  Before the change the what branch of the https://github.com/apache/nuttx/pull/12394/files#diff-9dfe671931738d39252fc8e902e892d172ac118551d14e1a454e324edb0d0be4R889-R897 would have been selected.

We have to be super carful here not break existing boards. That had the default set BEFORE this change

Reply:
I changed default value.